huxley/www/js/components/PaperSubmissionTable.js

```diff
@@ -24,11 +24,17 @@ class PaperSubmissionTable extends React.Component {
       var hrefData = files[paper.id]
         ? url.createObjectURL(files[paper.id])
         : null;
-      var gradedHrefData = graded_files[paper.id]
-        ? url.createObjectURL(graded_files[paper.id])
-        : null;
+      var gradedHrefData =  
+          paper.graded &&
+          paper.graded_file &&
+          graded_files[paper.id]
+            ? url.createObjectURL(graded_files[paper.id])
+            : null;
       var fileNames = paper.file.split("/");
       var fileName = fileNames[fileNames.length - 1];
+      var gradedFileNames = gradedHrefData ? paper.graded_file.split("/") : null;
+      var gradedFileName = gradedFileNames ? gradedFileNames[gradedFileNames.length - 1] : null;
```

Comment:
Over here, what if gradedFileNames is empty list, so `gradedFileName` is null then the next line we try to add "graded_" to null?

I don't think it will happen unless `paper.graded_file` is empty string or something similarly odd but the code just looks a bit funky.

Reply:
gradeFileName shouldn't be null if gradedHrefData isn't null so in this case it should be ok